### PR TITLE
(#1332) Add switch to disable logging for archive extraction

### LINKED ISSUE #1332: Install-ChocolateyPackage / Get-ChocolateyUnzip - switch to not log the files during unzip for zips with many files
Choco times-out for large zip files (`9, 500+` files). 

Perhaps an option to disable the logging of each file would speed-up the process?

(**Admin edit** - notes):
* Reference - [June 12, 2017 11:18 AM](https://gitter.im/chocolatey/choco?at=593ebec3c59bd9c4640bc4de)

**Admin edit** - Gitter screenshots
The above link to Gitter may disappear as we have transitioned to Discord for chat, so capturing some screenshots of the conversation:

### PR DESCRIPTION
**Pull request by TheCakeIsNaOH**
## Description Of Changes

Adds a switch "disableLogging" to Get-ChocolateyUnzip and
Install-ChocolateyZipPackage. This switch disable 7zip logging of
extracted filenames and does not write out the log of extracted
files. Install-ChocolateyZip passes the switch to Get-ChocolateyUnzip,
and the functionality is implemented into Get-ChocolateyUnzip.

## Motivation and Context

This is useful for large archives with a very large number of files as
logging the extracted files can significantly slow down the extraction
process.

## Testing

- Imported `chocolateyInstall.psm1` into a `powershell.exe -version 2` window, the module loads fine.
- Ran `.\choco.exe install filespy --verbose --debug` to validate that the files are still correctly logged into the debug output and to the file
- Downloaded the filespy package, edited the `chocolateyInstall.ps1` to add the `DisableLogging` parameter, and installed the edited package to validate that the switch is passed correctly from `Install-ChocolateyZipPackage` to `Get-ChocolateyUnzip` and to validate that the logging and output file are correctly disabled.

## Change Types Made

* [ ] Bug fix (non-breaking change)
* [x] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)

## Related Issue

Fixes #1332

## Change Checklist

* [x] Requires a change to the documentation
* [x] Documentation has been updated
* [ ] Tests to cover my changes, have been added
* [x] All new and existing tests passed.
* [x] PowerShell v2 compatibility checked.

### CODE REVIEW

**Comment by TheCakeIsNaOH** [comment]
Maybe I missed it, but I didn't see any tests that installed packages with archives, so there is nowhere to add a test at the moment.

**Comment by gep13** [comment]
Pulling this in would require an upstream change in the chocolatey.extension.  Going to hold off on this one for now...

**Comment by TheCakeIsNaOH** [comment]
> upstream change

Kind of pedantic, but wouldn't it be a downstream change? Since `chocolatey.extension` needs `chocolatey`, but not the other way arround?

File: src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1
Comment (gep13) [inline src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1:237]: @vexx32 am I right in saying that this is not the preferred way to do a `!` in PowerShell?

File: src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1
Comment (gep13) [inline src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1:172]:
```suggestion
  $loggingParam = '-bb1'
  if ($disableLogging) {
      $loggingParam = '-bb0'
  }
```

**Review by gep13 — CHANGES_REQUESTED** [review]

**Comment by gep13** [comment]
@TheCakeIsNaOH I have rebased this PR on the head of develop, so you will need to reset it before making any additional changes. I made my comments after rebasing as I thought I was ready to merge, and then I noticed a couple of things. Normally I don't force push to a branch unless I am ready to merge, apologies!

File: src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1
Comment (vexx32) [reply src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1:237]: in general I would tend to prefer `-not`, yes. Either works in practice, though.

Comment (TheCakeIsNaOH) [reply src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1]: Done.

File: src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1
Comment (TheCakeIsNaOH) [reply src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1]: Done.

**Comment by TheCakeIsNaOH** [comment]
@gep13 I've updated the PR with your suggested changes.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@TheCakeIsNaOH thanks for getting that fixed up!

**Pull request merged**
